internal/exec/stages/files/units.go

```diff
@@ -86,19 +86,36 @@ func (s *stage) createUnits(config types.Config) error {
 				}
 			}
 		}
-		if unit.Mask != nil && *unit.Mask {
-			relabelpath := ""
-			if err := s.Logger.LogOp(
-				func() error {
-					var err error
-					relabelpath, err = s.MaskUnit(unit)
+		if unit.Mask != nil {
+			if *unit.Mask { // mask: true
+				relabelpath := ""
+				if err := s.Logger.LogOp(
+					func() error {
+						var err error
+						relabelpath, err = s.MaskUnit(unit)
+						return err
+					},
+					"masking unit %q", unit.Name,
+				); err != nil {
 					return err
-				},
-				"masking unit %q", unit.Name,
-			); err != nil {
-				return err
+				}
+				s.relabel(relabelpath)
+			} else { // mask: false
+				masked, err := s.IsUnitMasked(unit)
+				if err != nil {
+					return err
+				}
```

Comment:
Any reason to not just drop this check & the subsequent `if masked` and just always call `s.UnmaskUnit`?

Reply:
Mostly because I don't want to litter the logs with `unmasking unit..` messages for things that don't need unmasking. Basically anyone specifying `mask: false` (which some tools specify for completeness when enabling another unit) will get that message in the logs. So I'm trying to make sure we only send that message to the logs if there is something to do.

See also the conversation in https://github.com/coreos/ignition/pull/1113#discussion_r510613663